### PR TITLE
Add {fmt} to DALI

### PR DESCRIPTION
**Pull request by klecki**
Change DALI_ENFORCE to use {fmt} internally

CUDA_LINK_LIBRARIES_KEYWORD cleanup

Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

File: dali/CMakeLists.txt [hunk 1]
```diff
@@ -88,6 +83,7 @@ target_link_libraries(${dali_lib} PRIVATE "-Wl,--exclude-libs,${exclude_libs}")
 # Link in dali's dependencies
 message(STATUS "Adding dependencies to ${dali_lib}: '${DALI_LIBS}'")
 target_link_libraries(${dali_lib} PRIVATE ${DALI_LIBS})
+target_link_libraries(${dali_lib} INTERFACE fmt::fmt)
```
Comment (mzient) [inline dali/CMakeLists.txt]: It's certainly not `INTERFACE`. `INTERFACE` means: "the target is not using the library itself, but the client will not be able to use the target without linking against that library".
We could even go for `PRIVATE` and instruct the user to link against `libfmt` if they need to use it in their code. Client code may even use `{fmt}` in header-only mode and need not link `libfmt` at all.

File: dali/kernels/CMakeLists.txt [hunk 2]
```diff
@@ -26,4 +26,4 @@ collect_test_sources(DALI_KERNEL_TEST_SRCS)
 cuda_add_library(${dali_kernel_lib} STATIC "${DALI_KERNEL_SRCS}")
 cuda_add_library(${dali_kernel_test_lib} STATIC "${DALI_KERNEL_TEST_SRCS}")
 
-target_link_libraries(${dali_kernel_test_lib} ${dali_kernel_lib})
+target_link_libraries(${dali_kernel_test_lib} PRIVATE ${dali_kernel_lib})
```
Comment (mzient) [inline dali/kernels/CMakeLists.txt]: Now you force the test binary to link `${dali_kernel_lib}` explicitly. I'd say it's `PUBLIC`.

**Comment by klecki** [comment]
!build

**Comment by szalpal** [comment]
How can it be used?

**Comment by klecki** [comment]
Won't do due to how symbols are propagated.

**Pull request closed without merging**